internal/build/tar_test.go

```diff
@@ -17,18 +17,22 @@ import (
 
 func TestArchiveDf(t *testing.T) {
 	f := newFixture(t)
-	ab := NewArchiveBuilder(model.EmptyMatcher)
-	defer ab.close()
-	defer f.tearDown()
 
 	dfText := "FROM alpine"
-	df := dockerfile.Dockerfile(dfText)
-	err := ab.archiveDf(f.ctx, df)
-	if err != nil {
-		t.Fatal(err)
-	}
+	pr, pw := io.Pipe()
```

Comment:
it would be a lot simpler to use a bytes.Buffer in all these tests. it should work fine as both a reader and a writer. and then you don't need the goroutine.

certainly, we should have at least one test that exercises concurrency, but you don't need all of them to do the concurrency dance.